src/Compilers/CSharp/Portable/CSharpResources.resx

```diff
@@ -5595,8 +5595,11 @@ To remove the warning, you can use /reference instead (set the Embed Interop Typ
   <data name="ERR_UnderscoreDeclaredAndDiscardPattern" xml:space="preserve">
     <value>The discard pattern '_' cannot be used where '{0}' is in scope.</value>
   </data>
-  <data name="ERR_ConstantPatternNamedUnderscore" xml:space="preserve">
-    <value>A constant named '_' cannot be used as a pattern.</value>
+  <data name="WRN_CaseConstantNamedUnderscore" xml:space="preserve">
+    <value>The name '_' refers to the constant '(0}', not the discard pattern. Use 'var _' to discard the value, or '@_' to refer to a constant by that name.</value>
```

Comment:
>'(0}' [](start = 47, length = 5)

`{0}`

Reply:
Why does the user need `@_`? If `_` always refers to the value named `_`, what use is the `@_` disambiguation?

Reply:
To suppress this warning.

---
In reply to: [230200698](https://github.com/dotnet/roslyn/pull/30847#discussion_r230200698) [](ancestors = 230200698)